tests/test_losses.py

```diff
@@ -69,6 +69,59 @@ def test_swav_loss(self):
         )
         _ = loss_layer(scores=self._get_embedding(), head_id=0)
 
+    def test_barlow_twins_loss(self):
+        loss_layer = BarlowTwinsCriterion(
+            lambda_=0.0051, scale_loss=0.024, embedding_dim=EMBEDDING_DIM
+        )
+        _ = loss_layer(self._get_embedding())
+
+
+class TestBarlowTwinsCriterion(unittest.TestCase):
+    """
+    Specific tests on Barlow Twins going further than just doing a forward pass
+    """
+
+    def test_barlow_twins_backward(self):
+        EMBEDDING_DIM = 3
+        criterion = BarlowTwinsCriterion(
+            lambda_=0.0051, scale_loss=0.024, embedding_dim=EMBEDDING_DIM
+        )
+        embeddings = torch.randn((4, EMBEDDING_DIM), requires_grad=True)
+
+        self.assertTrue(embeddings.grad is None)
+        criterion(embeddings).backward()
+        self.assertTrue(embeddings.grad is not None)
+        with torch.no_grad():
+            next_embeddings = embeddings - embeddings.grad  # gradient descent
+            self.assertTrue(criterion(next_embeddings) < criterion(embeddings))
+
+    @staticmethod
+    def worker_fn(gpu_id: int, world_size: int, batch_size: int):
+        dist.init_process_group(
+            backend="nccl",
+            init_method="tcp://0.0.0.0:1234",
```

Comment:
I was wondering if we can use https://github.com/facebookresearch/vissl/blob/master/vissl/utils/misc.py#L104 instead to construct the run_id ? the issue  with hardcoding the port 1234 is that sometimes the machines could have the ports being used already and so the test could work on some machines but on the other machines it wouldn't .